src/transformStateWithClones.js

```diff
@@ -6,8 +6,39 @@
  *
  * @return {Object[]}
  */
+const addProp = 'addProperties';
+const removeProp = 'removeProperties';
+const clear = 'clear';
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformedStates = [];
+  let stateCopy = { ...state };
+
+  for (const item of actions) {
+    switch (item.type) {
+      case addProp:
+        stateCopy = {
+          ...stateCopy, ...item.extraData,
+        };
+        break;
+
+      case removeProp:
+        for (const keys of item.keysToRemove) {
+          delete stateCopy[keys];
+        }
+        break;
+
+      case clear:
+        for (const keys in stateCopy) {
+          delete stateCopy[keys];
+        }
+        break;
+    }
```

Comment:
Checklist:
- [CODE STYLE]: switch/case should always have default case for error handling.